### PR TITLE
optimize rendering loop by avoiding creation of temporary Vector objects

### PR DESCRIPTION
**Pull request by uwemaurer**
I noticed with the Profiler in AndroidStudio that it creates a lot of Vector objects during rendering and then frequently runs the garbage collector. This change optimizes this.

### CODE REVIEW

**Comment by DanielMartinus** [comment]
Thanks for your contribution @uwemaurer! I see a lot of changes happening, could you maybe eleborate a little more what you did and how they all contribute to optimizing for garbage collecting?

File: konfetti/src/main/java/nl/dionsegijn/konfetti/Confetti.kt [hunk 1]
```diff
@@ -103,7 +100,9 @@ class Confetti(
             return
         }
 
-        paint.alpha = alpha
+        // setting alpha via paint.setAlpha allocates a temporary "ColorSpace$Named" object
+        // it is more efficient via setColor
```
Comment (DanielMartinus) [inline konfetti/src/main/java/nl/dionsegijn/konfetti/Confetti.kt:104]: Awesome, good to know!

**Review by uwemaurer — COMMENTED** [review]
I added some comments

File: konfetti/src/main/java/nl/dionsegijn/konfetti/Confetti.kt [hunk 2]
```diff
@@ -79,15 +75,16 @@ class Confetti(
...
+
     }
 
     private fun updateAlpha(deltaTime: Float) {
-        if (fadeOut) {
+        alpha = if (fadeOut) {
             val interval = 5 * deltaTime * speedF
-            if (alpha - interval < 0) alpha = 0
-            else alpha -= (5 * deltaTime * speedF).toInt()
+            (alpha - interval.toInt()).coerceAtLeast(0)
```
[the remark this answers is not in the document]
Comment (DanielMartinus) [reply konfetti/src/main/java/nl/dionsegijn/konfetti/Confetti.kt:85]: Love this improvement 👍

**Comment by DanielMartinus** [comment]
@uwemaurer didn't complete the review yet, I want to run some tests and see if I can see some noticeable changes. Will get back to it early next week

**Review by DanielMartinus — APPROVED** [review]
Thanks for the improvements @uwemaurer!

**Pull request merged**